http3/server.go

```diff
@@ -297,6 +322,15 @@ func (s *Server) serveListener(ln QUICEarlyListener) error {
 			return err
 		}
 		go func() {
+			s.connCount.Add(1)
+			defer func() {
+				s.mutex.RLock()
+				// close once when the server is closed and the connection count is zero
+				if s.connCount.Add(-1) == 0 && s.closed && s.doneChanClosed.CompareAndSwap(false, true) {
+					close(s.closeDoneChan)
+				}
+				s.mutex.RUnlock()
+			}()
```

Comment:
You're repeating a lot of code here. Let's just create a new method to avoid that.